### PR TITLE
Fix/dialog click event

### PR DESCRIPTION
**Pull request by Thienhuynh95**
## Issue

- Designer wants textarea to be able to adjust height to fit the content automatically.
- She also found the bug that the dialog will be closed if the mouse up or down is triggered on the dialog overlay.

https://github.com/Zonos/amino/assets/17950626/97cb66eb-7cfc-4898-9ea7-3581500415fd

## Description

This PR:
- add expandable feature for textarea [story url](https://amino-8pijpwbdn-zonos.vercel.app/?path=/docs/amino-textarea--docs) 
image

- fixes the bug on the dialog described above [story url](https://amino-8pijpwbdn-zonos.vercel.app/?path=/docs/amino-dialog--docs)

## Todo

- [ ] Bump version and add tag

### CODE REVIEW

File: src/components/dialog/_BaseDialog.tsx [hunk 1]
```diff
@@ -60,6 +60,7 @@ export const BaseDialog = ({
   width,
 }: BaseDialogProps) => {
   const { aminoTheme } = useAminoTheme();
+  const mouseDownTarget = useRef<HTMLDivElement | null>(null);
```
Comment (goodpickle) [inline src/components/dialog/_BaseDialog.tsx:63]: What do you think about removing the `| null`? I believe Ref's always allow `null` 🤔 

```suggestion
  const mouseDownTarget = useRef<HTMLDivElement>(null);
```

**Review by goodpickle — CHANGES_REQUESTED** [review]

File: src/components/textarea/Textarea.tsx [hunk 2]
```diff
@@ -158,16 +181,30 @@ export const Textarea = forwardRef<HTMLTextAreaElement, TextareaProps>(
     ref
   ) => {
     const hasValue = !!value;
+    const textareaRef = useRef<HTMLTextAreaElement | null>(null);
```
[the remark this answers is not in the document]
Comment (Thienhuynh95) [reply src/components/textarea/Textarea.tsx:184]: I think the `null` union is necessary when I want to re-assign the  ref with other ref (this case is the one from fowardRef).

**Comment by evad1n** [comment]
That outside click dismiss can be disabled with props `closeOnBlur` I believe.

**Comment by Thienhuynh95** [comment]
@evad1n The dialog works as intented, but the close behavior is not. We want to close the dialog when clicking outside of the dialog area, but also don't want to close it when you are clicking inside and dragging it outside of the dialog.

**Comment by evad1n** [comment]
Would changing the `onClick` to `onMouseDown` fix the issue?

**Review by goodpickle — APPROVED** [review]

**Comment by Thienhuynh95** [comment]
@evad1n With just `keyDown` will not fix the scenario where you click outside and then drag into the dialog.

**Pull request merged**